### PR TITLE
Add Node.js client codes for entry listeners with predicates [API-1484]

### PR DESCRIPTION
**Pull request by fatihozer0**

### CODE REVIEW

File: docs/modules/data-structures/pages/listening-for-map-entries.adoc [hunk 1]
```diff
@@ -55,20 +55,31 @@ Node.js::
...
+    const surname;
+
+    constructor(surname) {
+        this.surname = surname;
+    }
+
+    toString() {
+        return "Employee: " + this.surname;
+    }
+}
```
Comment (srknzl) [inline docs/modules/data-structures/pages/listening-for-map-entries.adoc:69]: This is not valid Node.js:

image

File: docs/modules/data-structures/pages/listening-for-map-entries.adoc [hunk 2]
```diff
@@ -55,20 +55,31 @@ Node.js::
...
 ----
 --
-Python:: 
-+ 
--- 
+Python::
++
+--
```
Comment (srknzl) [inline docs/modules/data-structures/pages/listening-for-map-entries.adoc:74]: Please revert these unnecessary changes.

File: docs/modules/data-structures/pages/listening-for-map-entries.adoc [hunk 3]
```diff
@@ -99,21 +110,37 @@ C++::
...
+    added: (entryEvent) => {
+        console.log('Entry added: ' + entryEvent.value);
+    },
+    removed: (entryEvent) => {
+        console.log('Entry removed: ' + entryEvent.value);
+    },
+    updated: (entryEvent) => {
+        console.log('Entry updated: ' + entryEvent.value);
+    }
+}
```
Comment (srknzl) [inline docs/modules/data-structures/pages/listening-for-map-entries.adoc:140]:
```suggestion
};
```

File: docs/modules/data-structures/pages/listening-for-map-entries.adoc [hunk 4]
```diff
@@ -99,21 +110,37 @@ C++::
...
+await map.addEntryListenerWithPredicate(listener, new Predicates.sql('surname = smith'), undefined, true);
```
Comment (srknzl) [inline docs/modules/data-structures/pages/listening-for-map-entries.adoc:141]:
```suggestion
await map.addEntryListenerWithPredicate(listener, new Predicates.sql('surname=smith'), undefined, true);
```

File: docs/modules/data-structures/pages/listening-for-map-entries.adoc [hunk 5]
```diff
@@ -99,21 +110,37 @@ C++::
...
+--
 [source,javascript]
 ----
-
+const client = await Client.newHazelcastClient();
+const map = await client.getMap('map');
+
+const listener = {
+    added: (entryEvent) => {
+        console.log('Entry added: ' + entryEvent.value);
```
Comment (srknzl) [inline docs/modules/data-structures/pages/listening-for-map-entries.adoc:132]: technically value is not entry. can you print both key and value with something like: (in backticks) `key: ${entryEvent.key}, value: ${entryEvent.value}`

File: docs/modules/data-structures/pages/listening-for-map-entries.adoc [hunk 6]
```diff
@@ -99,21 +110,37 @@ C++::
...
-Node.js:: 
-+ 
--- 
+Node.js::
++
+--
 [source,javascript]
 ----
-
+const client = await Client.newHazelcastClient();
```
Comment (srknzl) [inline docs/modules/data-structures/pages/listening-for-map-entries.adoc:127]: One thing I noticed, Java provides whole working code sample. Can you also do the same? Example:

```js
const { Client, Predicates } = require('hazelcast-client');

async function main() {
    const client = await Client.newHazelcastClient();
    const map = await client.getMap('map');
    const listener = {
        added: (entryEvent) => {
            console.log('Entry added: ' + entryEvent.value);
        },
        removed: (entryEvent) => {
            console.log('Entry removed: ' + entryEvent.value);
        },
        updated: (entryEvent) => {
            console.log('Entry updated: ' + entryEvent.value);
        }
    };
    await map.addEntryListenerWithPredicate(listener, new Predicates.sql('surname = smith'), undefined, true);
    console.log('Entry listener registered');
    await client.shutdown();
}

main().catch(err => {
    console.error('Error occurred:', err);
    process.exit(1);
});

```

**Review by srknzl — CHANGES_REQUESTED** [review]
Requested some changes.

**Comment by srknzl** [comment]
@fatihozer0 These 85 commits from several people looks problematic. Can you extract the required commits and only add them to this PR?

**Comment by fatihozer0** [comment]
PR is closed because of a mistake I've done. Continues here #520

**Pull request closed without merging**